webapp/src/components/hooks/survey.ts

```diff
@@ -3,56 +3,23 @@
 
 import {useCallback, useEffect, useRef, useState} from 'react';
 
-import type {SurveyResponse, UserSurvey} from 'types/plugin';
+import type {Post} from '@mattermost/types/posts';
 
-export function useUserSurvey(surveyID: string) {
+import type {Survey, SurveyResponse, UserSurvey} from 'types/plugin';
+
+export function useUserSurvey(post: Post) {
     const [survey, setSurvey] = useState<UserSurvey>();
     const linearScaleQuestionID = useRef<string>();
 
     const responsesExist = survey?.response !== undefined;
     const surveyExpired = survey?.status === 'ended';
 
-    const fetchUserSurvey = async (surveyID: string) => {
-        // make API call here to get user survey...
-        const survey: UserSurvey = {
-            surveyId: surveyID,
-            startDate: '01/04/2024',
-            endDate: '30/04/2024',
-            questions: [
-                {
-                    id: '49fc9a85-b4d8-424e-b13f-40344b168123',
-                    mandatory: true,
-                    system: true,
-                    text: 'How likely are you to recommend Mattermost?',
-                    type: 'linear_scale',
-                },
-                {
-                    id: 'aa026055-c97c-48d7-a025-c44590078963',
-                    mandatory: true,
-                    system: true,
-                    text: 'How can we make your experience better?',
-                    type: 'text',
-                },
-                {
-                    id: '0eee4429-5083-41ef-bda7-2ee9c5ece929',
-                    mandatory: false,
-                    system: false,
-                    text: 'option question text',
-                    type: 'text',
-                },
-            ],
-            status: 'in_progress',
-
-            // response: {
-            //     responses: {
-            //         '49fc9a85-b4d8-424e-b13f-40344b168123': '8',
-            //         'aa026055-c97c-48d7-a025-c44590078963': 'Response 1',
-            //         '0eee4429-5083-41ef-bda7-2ee9c5ece929': 'Response 2',
-            //     },
-            //     dateCreated: format(new Date(), 'dd/MM/yyyy'),
-            // },
-        };
+    useEffect(() => {
+        if (!post.props.survey_questions) {
+            return;
+        }
 
+        const survey = JSON.parse(post.props.survey_questions) as Survey;
```

Comment:
This is a security issue waiting to happen.

Post props are not validated around the app, so we need to add some validation to `survey` so there is not a crash when trying to access its values.

I would say the simplest is just to consider this instead of Survey, consider it `Partial<Survey>`. But we could do some more in depth verification.

Also... JSON.parse can throw an exception if the string is not a valid JSON, so it would be better to wrap it on a try catch.

Reply:
Done, added generated type checking code in `validators.ts`.

Reply:
@larkox I actually added the check in the other PR as there were a lot of  changes with the hook in that PR and it avoided re-doing some work. Its in #13 validators.ts -

Reply:
apologies for the confistion.